emcc.py

```diff
@@ -1010,7 +1010,8 @@ def add_link_flag(i, f):
         file_suffix = get_file_suffix(arg)
         if file_suffix in SOURCE_ENDINGS + OBJECT_FILE_ENDINGS + DYNAMICLIB_ENDINGS + ASSEMBLY_ENDINGS + HEADER_ENDINGS or shared.Building.is_ar(arg): # we already removed -o <target>, so all these should be inputs
           newargs[i] = ''
-          if file_suffix.endswith(SOURCE_ENDINGS):
+          #if file_suffix.endswith(SOURCE_ENDINGS):
+          if file_suffix.endswith(SOURCE_ENDINGS) or (has_dash_c and file_suffix.endswith(OBJECT_FILE_ENDINGS)):
```

Comment:
Hmm.. my outstanding change is about to get rid of OBJECT_FILE_ENDINGS.   Should this perhaps be "is_bitcode"? or check bitcode endings specifically?   We should do whatever clang does I guess.  Does it allow `.o` as an input to compiler, or just `.bc`?
 
See https://github.com/emscripten-core/emscripten/pull/10560/

Reply:
It seems to only accept .bc, for .o, it says:
clang-11: warning: hello.exe.o: 'linker' input unused [-Wunused-command-line-argument]

Reply:
In that case maybe just use ".bc" rather than OBJECT_FILE_ENDINGS?

Reply:
Done.